### ISSUE TITLE
[Accessibility] Narrator screen reader blue rectangle is wide for a new character in ComboBox and Narrator will focus on the Combobox(RightToLeft) control rather than the new character when the length of the characters is greater than the width of the control

### ISSUE THREAD

**Issue by John-Qiao** [dotnet/winforms#4079]
* .NET Core Version: 
.NET SDK 5.0.100-rtm.20508.17

* Have you experienced this same bug with .NET Framework?: 
Yes

**More Info::**

1. This issue occurs in .NET Framework too.
2. This issue occurs in the standard ComboBox control and the ToolStripComboBox/MenuStripComboBox items. 

**Problem description:**

1. In step 2: Narrator screen reader blue rectangle is wide for a new character. 
![Actual-1](https://user-images.githubusercontent.com/45864985/95568403-c2681b80-0a56-11eb-8abf-b1c2a8a461c1.png)

2. In step 3: Narrator focus on the Combobox(RightToLeft) control rather than the new character.
![Actual-2](https://user-images.githubusercontent.com/45864985/95568408-c5630c00-0a56-11eb-8047-126fa75081da.png)

**Expected behavior:**

1. In step 2: The Narrator screen reader blue rectangle should be thin like it works in the TextBox control. 
![Expected-1](https://user-images.githubusercontent.com/45864985/95568414-c85dfc80-0a56-11eb-810e-226adc339c82.png)

2. In step 3: Narrator should focus on the new character like it works in the Combobox(LeftToRight) control. 
![Expected-2](https://user-images.githubusercontent.com/45864985/95568422-cac05680-0a56-11eb-8573-998e46ade1e4.png)

**Minimal repro:**
[TestApp.zip](https://github.com/dotnet/winforms/files/5353981/TestApp.zip)
1. Open the attached TestApp project in VS, run it and launch the Narrator.
2. Focus on one of the ComboBox control(no matter which one), type a character, for example 'd', and observe the width of Narrator screen reader blue rectangle on 'd'.
3. Focus on the ComboBox(RightToLeft) control, then type characters continuously in it until the length of the characters is greater than the width of the control, observe the Narrator screen reader blue rectangle on ComboBox(RightToLeft) control.  
![ReproSteps](https://user-images.githubusercontent.com/45864985/95568386-bc723a80-0a56-11eb-8c19-b8e78146729c.gif)

**Comment by merriemcgaw** [comment]
@vladimir-krestov Let's recheck this and discuss at our next triage. We'll decide what the best way is to manage the ComboBox Text Pattern in general.

**Comment by merriemcgaw** [comment]
I'd like the TextPattern to behave as it does in the TextBox if possible.

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. I think you are right and adding support of the text pattern fixed this issue. Now I see the correct behavior of the `ComboBox` for both cases:

![4079](https://user-images.githubusercontent.com/23376742/128683289-5d89663f-3ae6-4076-ab4c-e22a3f556654.gif)

**Comment by John-Qiao** [comment]
Verified with .NET 6.0.100-rc.1.21410.51 from main branch, this issue has been fixed that the size of Narrator screen reader blue rectangle and behavior of focus on the new character in ComboBox control or item are both correctly now.
